data-prepper-plugins/kinesis-source/src/test/java/org/opensearch/dataprepper/plugins/kinesis/source/processor/KinesisRecordProcessorTest.java

```diff
@@ -0,0 +1,371 @@
+
+        Thread.sleep(2000);
+
+        kinesisRecordProcessor.processRecords(processRecordsInput);
+
+        verify(checkpointer).checkpoint();
+        verify(buffer).writeAll(anyCollection(), anyInt());
+        verify(acknowledgementSetManager, times(1)).create(any(), any(Duration.class));
+    }
+
+    @Test
+    void testProcessRecordsWithAcknowledgementsEnabledAndAcksReturnFalse()
+            throws Exception {
+        List<KinesisClientRecord> kinesisClientRecords = createInputKinesisClientRecords();
+        when(processRecordsInput.records()).thenReturn(kinesisClientRecords);
+        when(kinesisSourceConfig.isAcknowledgments()).thenReturn(true);
+        when(kinesisStreamConfig.isEnableCheckPoint()).thenReturn(false);
+        AtomicReference<Integer> numEventsAdded = new AtomicReference<>(0);
+        doAnswer(a -> {
+            numEventsAdded.getAndSet(numEventsAdded.get() + 1);
+            return null;
+        }).when(acknowledgementSet).add(any());
+
+        doAnswer(invocation -> {
+            Consumer<Boolean> consumer = invocation.getArgument(0);
+            consumer.accept(false);
+            return acknowledgementSet;
+        }).when(acknowledgementSetManager).create(any(Consumer.class), any(Duration.class));
+
+        kinesisRecordProcessor = new KinesisRecordProcessor(buffer, kinesisSourceConfig, acknowledgementSetManager, pluginMetrics, pluginFactory, streamIdentifier);
+        kinesisRecordProcessor.initialize(initializationInput);
+
+        Thread.sleep(2000);
+
+        kinesisRecordProcessor.processRecords(processRecordsInput);
+
+        verify(checkpointer, times(0)).checkpoint();
+        verify(buffer).writeAll(anyCollection(), anyInt());
+        verify(acknowledgementSetManager, times(1)).create(any(), any(Duration.class));
+    }
+
+    @Test
+    void testProcessRecordsWithNDJsonInputCodec()
+            throws Exception {
+        when(processRecordsInput.records()).thenReturn(createInputKinesisClientRecords());
+        when(kinesisSourceConfig.isAcknowledgments()).thenReturn(false);
+        when(kinesisStreamConfig.isEnableCheckPoint()).thenReturn(true);
+
+        PluginModel pluginModel = mock(PluginModel.class);
+        when(pluginModel.getPluginName()).thenReturn("ndjson");
+        when(pluginModel.getPluginSettings()).thenReturn(Collections.emptyMap());
+        when(kinesisSourceConfig.getCodec()).thenReturn(pluginModel);
+
+        InputCodec codec = mock(InputCodec.class);
+        when(pluginFactory.loadPlugin(eq(InputCodec.class), any())).thenReturn(codec);
+
+        when(acknowledgementSetManager.create(any(), any(Duration.class))).thenReturn(acknowledgementSet);
+        kinesisRecordProcessor = new KinesisRecordProcessor(buffer, kinesisSourceConfig, acknowledgementSetManager, pluginMetrics, pluginFactory, streamIdentifier);
+        kinesisRecordProcessor.initialize(initializationInput);
+
+        Thread.sleep(2000);
```

Comment:
Use Awaitility instead of a sleep.

Reply:
I have modified the test to remove this.